### PR TITLE
Update metamask

### PR DESCRIPTION
**Pull request by r3kt-eth**
hi,

1) updated metamask to new version
2) updated all flows to work with new version
3) fixed waitToBeHidden and added it to waitUntilStable
4) handled critical metamask error
5) updated documentation
6) updated trunk

this pull request fix critical issue with notifications in background too.

thx, cheerz.

### CODE REVIEW

**Comment by drptbl** [comment]
Hey,

Thanks for contribution. I will review it asap.

Cheers,
Jakub.

File: commands/metamask.js [hunk 1]
```diff
@@ -765,6 +786,15 @@ module.exports = {
   },
   confirmPermissionToSpend: async () => {
     const notificationPage = await playwright.switchToMetamaskNotification();
+    await playwright.waitAndSetValue(
+      '999999999999999999',
```
Comment (drptbl) [inline commands/metamask.js:790]: can you please make this as an (optional) parameter?

File: pages/metamask/first-time-flow-page.js [hunk 2]
```diff
@@ -1,77 +1,90 @@
...
+// const newSignupCheckbox = `${firstTimeFlowCreatePage} .first-time-flow__checkbox`;
+// const createButton = `${firstTimeFlowCreatePage} .first-time-flow__button`;
+// module.exports.firstTimeFlowCreatePagePageElements = {
+//   firstTimeFlowCreatePage,
+//   newPasswordInput,
+//   confirmNewPasswordInput,
+//   newSignupCheckbox,
+//   createButton,
+// };
+
```
Comment (drptbl) [inline pages/metamask/first-time-flow-page.js:57]: please remove this if it's not needed anymore

**Review by drptbl — CHANGES_REQUESTED** [review]
Added few comments. Also, `importWallet` and `createWallet` looks very similar now and could be extracted to separate function instead of repeating it. Thanks 👍.

**Review by drptbl — APPROVED** [review]
LGTM

**Comment by ComradeAERGO** [comment]
Didn't this breaks the tests?
It seems like the cy.get(#network) calls from the metamask-spec.js file all fail now.

**Pull request merged**